### PR TITLE
feat: [CDS-71966]: Create instance stats delete job to delete data for 2 months old accounts

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
creating a quartz job, InstanestatsDelete job to delete instance_stats present in timescaleDB for accounts that have expired 2months ago

## Testing
1. when timescaleDBconnectioon not there
2. retry working correctly
3. deletion of all data regarding an account
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50827)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/scheduler/InstanceStatsDeleteJob.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import org.quartz.JobBuilder;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.SimpleTrigger;
+import org.quartz.TriggerBuilder;
+
+@DisallowConcurrentExecution
+@Slf4j
+public class InstanceStatsDeleteJob implements Job {
```
Comment (abosii-harness) [inline 400-rest/src/main/java/software/wings/scheduler/InstanceStatsDeleteJob.java:50]: Any reasons why this is part of the cg manager?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 400-rest/src/main/java/software/wings/scheduler/InstanceStatsDeleteJob.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  @Inject private PersistentLocker persistentLocker;
+
+  @Inject private AccountService accountService;
+  @Inject private FeatureFlagService featureFlagService;
+
+  private static TriggerBuilder<SimpleTrigger> instanceStatsTriggerBuilder(String accountId) {
+    return TriggerBuilder.newTrigger()
+        .withIdentity(accountId, GROUP)
+        .withSchedule(SimpleScheduleBuilder.simpleSchedule()
+                          .withIntervalInHours(SYNC_INTERVAL)
```
[the remark this answers is not in the document]
Comment (TarunUba1) [reply 400-rest/src/main/java/software/wings/scheduler/InstanceStatsDeleteJob.java:82]: i can do for running every week at sunday/ saturday or every month
cant do it bi-weekly

File: 400-rest/src/main/java/software/wings/scheduler/InstanceStatsDeleteJob.java (hunk 1 above)
Comment (TarunUba1) [reply 400-rest/src/main/java/software/wings/scheduler/InstanceStatsDeleteJob.java:50]: Cause this is for CG stats only

File: 390-db-migration/src/main/java/io/harness/migrations/all/AddInstanceStatsDeletionJobToAllAccounts.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import software.wings.scheduler.InstanceStatsDeleteJob;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import dev.morphia.query.Query;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AddInstanceStatsDeletionJobToAllAccounts implements Migration {
+  @Inject private WingsPersistence wingsPersistence;
```
[the remark this answers is not in the document]
Comment (TarunUba1) [reply 390-db-migration/src/main/java/io/harness/migrations/all/AddInstanceStatsDeletionJobToAllAccounts.java:25]: didn't get this comment

**Pull request closed without merging**